KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+| getPastEvent(event: String): List&#60;Object&#62; | Gets past events for this contract. |
+| getPastEvent(event: String, options: Object): List&#60;Object&#62; | Gets past events for this contract. The options object can define the filter options needed when calling [klay_logs]. See [klay_logs] for more details about options. |
+| getPastEvent(event: String, callback: Function): List&#60;Object&#62; | Gets past events for this contract. |
+| getPastEvent(event: String, options: Object, callback: Function): List&#60;Object&#62; | Gets past events for this contract. The options object can define the filter options needed when calling [klay_logs]. See [klay_logs] for more details about options. |
+
+#### ContractMethod
+
+`ContractMethod` is a class that contains abi information of smart contract function.
+
+- Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| name: String | The name of the function in smrat contract. |
+| inputs: List&#60;[ContractIOType](#contractiotype)&#62; | The input values of the function. In the list, each parameter of the function is defined as ContractIOType. When the `call` or `send` function is called, this is used to encode the parameter to create an input string. |
+| outputs: List&#60;[ContractIOType](#contractiotype)&#62; | The output values of the function. This is used to decode the value returned as the result of executing the function. |
+| signature: String | The [function signature](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.contract#cf-function-signature-function-selector) (function selector). The first four bytes of the input data for specifying the function to be called (or executed). It is the first (left, high-order in big-endian) four bytes of the Keccak-256 (SHA-3) hash of the signature of the function. |
+| nextMethods: List&#60;ContractMethod&#62; | nextMethods stores functions with the same name implemented in smart contracts. If the parameter passed by the user is different from the input of this contractMethod, it traverses the contractMethods defined in nextMethods to find the contractMethod to be called. |
+
+- Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| call(argumetns: List&#60;any&#62;, callObject: Object): any | Call a "constant" method and execute its smart contract method in the Klaytn Virtual Machine without sending any transaction. See [klay_call] for more details about callObject. |
+| send(arguments: List&#60;any&#62;): Object | Send a transaction to the smart contract and execute its method. This can alter the smart contract state. Send a transaction using the value defined in defaultSendOptions of the Contract. It thorws an exception if the value required for transaction (i.e `from` or `gas`) is not defined in defaultSendOptions. |
```

Comment:
```suggestion
| send(arguments: List&#60;any&#62;): Object | Send a transaction to the smart contract and execute its method. This can alter the smart contract state. Send a transaction using the value defined in defaultSendOptions of the Contract. It throws an exception if the value required for transaction (i.e `from` or `gas`) is not defined in defaultSendOptions. |
```